plugins/thunderbolt/fu-thunderbolt-firmware.c

```diff
@@ -734,54 +749,59 @@ fu_thunderbolt_image_validate (GBytes  *controller_fw,
 		};
 		locations = drom_locations;
 		if (!compare_locations (&locations, &controller, &image, error))
-			return VALIDATION_FAILED;
+			return FALSE;
 	}
 
 	if (!compare_pd_existence (hw_info->id, &controller, &image, error))
-		return VALIDATION_FAILED;
+		return FALSE;
 
 	/*
 	 * 0 is for the unknown device case, for being future-compatible with
 	 * new devices; so we can't know which locations to check besides the
 	 * vendor and model IDs that were validated already, but those should be
 	 * good enough validation.
 	 */
-	if (hw_info->id == 0)
-		return UNKNOWN_DEVICE;
+	if (hw_info->id == 0) {
+		g_set_error_literal (error,
+				     G_IO_ERROR,
+				     G_IO_ERROR_NOT_SUPPORTED,
+				     "no supported");
+		return FALSE;
```

Comment:
So we decided we don't want to support newer devices if the host support hasn't added yet?
At least let's remove the comment that says we do want to support it.